network.c

```diff
@@ -1072,6 +1072,10 @@ struct iio_context * network_create_context(const char *host)
 	char *description, *uri, *end, *port = NULL;
 	char port_str[6];
 	uint16_t port_num = IIOD_PORT;
+	char host_buf[FQDN_LEN + sizeof(":65535") + 1];
+	char *host = hostname ? host_buf : NULL;
+
+	iio_strlcpy(host_buf, hostname, sizeof(host_buf));
```

Comment:
love the first sentence :smile:

Reply:
I mean, who thought it was a good idea to return a non-const pointer to an immutable string?

Reply:
yeah, great API design...